delta/plugins/statistics/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/statistics/model/PropertiesStatistics.scala

```diff
@@ -0,0 +1,102 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.statistics.model
+
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.plugins.statistics.Statistics.{name, toPaths}
+import ch.epfl.bluebrain.nexus.delta.plugins.statistics.model.PropertiesStatistics.Metadata
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.ContextValue
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.deriveConfiguredEncoder
+import io.circe._
+import io.circe.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.plugins.statistics.contexts
+
+import scala.annotation.{nowarn, tailrec}
+
+/**
+  * Each property with its counts.
+  *
+  * @param metadata   the property information (names and counts)
+  * @param properties the nested properties of the current property
+  */
+final case class PropertiesStatistics(metadata: Metadata, properties: Seq[PropertiesStatistics])
```

Comment:
I still feel like a similar approach to the one we adopted for relationships (declare the properties and the relationships between them separately) would make things simpler

Reply:
What do you mean? This case class is just dealing with properties, as defined [here](https://github.com/BlueBrain/nexus/issues/2544#issuecomment-859624949). It has nothing to do with relationships...

Reply:
Sorry, I wasn't very clear in my previous comment
I was meaning that if we add a different contract where we first define the different properties and in a second part the parent child link (as we do in a relationship contract), we would not need recursion and it should be less complex and faster to evaluate especially for large resources as we could do it in one global pass.

Reply:
Both contracts have been agreed upon between frontend and backend. Doesn't seem like we should change that now

Reply:
It is just a comment sharing a thought, we don't have to change this now.